distributed/scheduler.py

```diff
@@ -8205,6 +8202,7 @@ def decide_worker(
         candidates = set(all_workers)
     else:
         candidates = {wws for dts in ts.dependencies for wws in dts.who_has}
+        candidates &= all_workers
```

Comment:
This fixes #8019

Reply:
FWIW I think this is a situation where an actual `decide_worker` unit test would be appropriate